<launch>

	<!-- knowledge base -->
	<node name="rosplan_knowledge_base" pkg="squirrel_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
	</node>

	<!-- scene database (MongoDB) -->
	<node name="rosplan_scene_database" pkg="mongodb_store" type="mongodb_server.py" respawn="false" output="screen">
	    <param name="database_path" value="$(find squirrel_knowledge_base)/common/mongoDB" />
	    <param name="mongodb_port" value="27020" />
	</node>
	<node name="rosplan_scene_message_store" pkg="mongodb_store" type="message_store_node.py" respawn="false" output="screen">
	    <param name="mongodb_port" value="27020" />
	</node>

	<!-- knowledge interfaces -->
	<node name="rosplan_roadmap_server" pkg="squirrel_knowledge_base" type="rproadmapServer" respawn="false" output="screen"></node>
	<node name="rosplan_object_server" pkg="squirrel_knowledge_base" type="rpobjectServer" respawn="false" output="screen"></node>

	<!-- planning system -->
	<node name="rosplan_planning_system" pkg="squirrel_planning_system" type="planner" respawn="false" output="screen">

		<!-- directory for generated files -->
	    <param name="data_path" value="$(find squirrel_planning_system)/common/" />
	    <param name="domain_path" value="$(find squirrel_planning_system)/common/domain_push.pddl" />
	    <param name="problem_path" value="$(find squirrel_planning_system)/common/problem.pddl" />

		<!-- to run the planner -->
	    <param name="planner_command" value="timeout 10 $(find squirrel_planning_system)/common/bin/popf -n" />
	</node>

	<!-- action interfaces -->
	<node name="rosplan_interface_movebase" pkg="squirrel_planning_system" type="rpmovebase" respawn="false" output="screen"></node>
	<node name="rosplan_interface_perception" pkg="squirrel_planning_system" type="rpperceptionaction" respawn="false" output="screen"></node>
	<node name="rosplan_interface_push" pkg="squirrel_planning_system" type="rppushaction" respawn="false" output="screen"></node>

</launch>
